pages/api/id/[form]/submission/confirm.ts

```diff
@@ -125,24 +125,17 @@ async function getSubmissionsFromConfirmationCodes(
       const submission = accumulatedSubmissions[currentConfirmationCode];
 
       if (!submission) {
-        return {
-          ...acc,
-          confirmationCodesNotFound: acc.confirmationCodesNotFound.concat(currentConfirmationCode),
-        };
+        acc.confirmationCodesNotFound.push(currentConfirmationCode);
+        return acc;
       } else if (submission.removalDate) {
-        return {
-          ...acc,
-          confirmationCodesAlreadyUsed:
-            acc.confirmationCodesAlreadyUsed.concat(currentConfirmationCode),
-        };
+        acc.confirmationCodesAlreadyUsed.push(currentConfirmationCode);
+        return acc;
       } else {
-        return {
-          ...acc,
-          submissionsToConfirm: acc.submissionsToConfirm.concat({
-            name: submission.name,
-            confirmationCode: currentConfirmationCode,
-          }),
-        };
+        acc.submissionsToConfirm.push({
+          name: submission.name,
+          confirmationCode: currentConfirmationCode,
+        });
+        return acc;
```

Comment:
@bryan-robitaille I always have this functional/immutability mindset 😆 . We this change I am pretty sure we can even call `return acc` just once at the end of the `if/elseif/else` block.